src/main/java/com/okta/developer/security/oauth2/ScimUserService.java

```diff
@@ -107,6 +107,7 @@ public ScimUser create(ScimUser resource) throws UnableToCreateResourceException
         if (user.isPresent()) {
             throw new UnableToCreateResourceException(Response.Status.CONFLICT, "User '" + resource.getUserName() + "' already exists.");
         } else {
+            resource.setId(UUID.randomUUID().toString()); // TODO: this should probably be configured in the persistence layer?
```

Comment:
This is 🤮, but I didn't want to dig into this, I'm guessing you could tweak it.
The SCIM server doesn't set an ID on create, so the SCIMple server will need to do it.

Reply:
In the previous implementation, I used the "sub" claim, but I guess that's not available with SCIM.